### PR TITLE
Add framework, services and init vuex store

### PR DESCRIPTION
**Pull request by anthonkendel**
Commit will add the following:
* http servicce
* game service
* rule service
* requirement service
* axios library for requests
* setup vuex store

Commit will change the following:
* update game model as it did not work properly with the response from
firebase
* change game view to display a full json game response
* change route to use uuid instead of id
* update game list to use data from vuex store
* move models into a folder called common

### CODE REVIEW

File: frontend/src/components/DgGamesList.vue [hunk 1]
```diff
@@ -40,25 +31,29 @@
...
+        filteredGames: 'getFilteredGames',
+        rules: 'getRules',
+        requirements: 'getRequirements',
+      }),
+    },
+    filters: {
+      toRating(array) {
+        if (!Array.isArray(array)) return -1;
+        const total = array.reduce((a, b) => a + b, 0);
+        const rating = total / array.length || -1;
```
Comment (naxxatou) [inline frontend/src/components/DgGamesList.vue]: If value is NaN it should be set to 0 instead, not -1

File: frontend/src/components/DgGamesList.vue [hunk 2]
```diff
@@ -40,25 +31,29 @@
...
+      ...mapGetters({
+        games: 'getGames',
+        filteredGames: 'getFilteredGames',
+        rules: 'getRules',
+        requirements: 'getRequirements',
+      }),
+    },
+    filters: {
+      toRating(array) {
+        if (!Array.isArray(array)) return -1;
```
Comment (naxxatou) [inline frontend/src/components/DgGamesList.vue]: Would prefer the expanded version of if statements

File: frontend/src/common/services/gameService.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import Game from '@/common/models/Game';
+import httpService from './httpService';
+import ruleService from './ruleService';
+import requirementService from './requirementService';
+
+async function getAllRules(array) {
```
Comment (naxxatou) [inline frontend/src/common/services/gameService.js]: Use Promise.all instead in this case, will shorten the code and make it more readable

Comment (anthonkendel) [reply frontend/src/common/services/gameService.js]: Fixed `getAllRules` and `getAllRequirements`, changed parameter to `ruleUuids` and `requirementUuids`

File: frontend/src/components/DgGamesList.vue (hunk 2 above)
Comment (anthonkendel) [reply frontend/src/components/DgGamesList.vue]: Fixed

File: frontend/src/components/DgGamesList.vue (hunk 1 above)
Comment (anthonkendel) [reply frontend/src/components/DgGamesList.vue]: Changed to 0

**Pull request merged**